plugins/UM3NetworkPrinting/resources/qml/PrinterVideoStream.qml

```diff
@@ -34,22 +34,29 @@ Item {
 
     Cura.NetworkMJPGImage {
         id: cameraImage
-        anchors.horizontalCenter: parent.horizontalCenter;
-        anchors.verticalCenter: parent.verticalCenter;
-        height: Math.round((imageHeight === 0 ? 600 * screenScaleFactor : imageHeight) * width / imageWidth);
+        anchors.horizontalCenter: parent.horizontalCenter
+        anchors.verticalCenter: parent.verticalCenter
+
+        readonly property real img_scale_factor: {
+            if (imageWidth > maximumWidth || imageHeight > maximumHeight) {
```

Comment:
Just to double-check: There's no way _both_ of these will be `inf`, `nan` or `0.0` right?

Reply:
Both `imageWidth` and `imageHeight` _might_ be 0 (initially they are always 0 because the image here is only initialized, not loaded https://github.com/Ultimaker/Cura/blob/0e1262126a4400633c2a57f7a1f2e230b7c36fd4/cura/PrinterOutput/NetworkMJPGImage.py#L79). They can never be Inf or NaN

Reply:
Both `imageWidth` and `imageHeight` _might_ be 0 (initially they are always 0 because the image here is only initialized, not loaded here)
 https://github.com/Ultimaker/Cura/blob/0e1262126a4400633c2a57f7a1f2e230b7c36fd4/cura/PrinterOutput/NetworkMJPGImage.py#L78-L80
 They can never be Inf or NaN